src/server/service/kv.rs

```diff
@@ -1558,18 +1558,28 @@ fn future_raw_batch_put<E: Engine, L: LockManager>(
     mut req: RawBatchPutRequest,
 ) -> impl Future<Output = ServerResult<RawBatchPutResponse>> {
     let cf = req.take_cf();
+    let pairs_len = req.get_pairs().len();
     let pairs = req
         .take_pairs()
         .into_iter()
         .map(|mut x| (x.take_key(), x.take_value()))
         .collect();
+    // ttls param is option, and it isn't repeated in old version,
+    // so we need to make code be compatible with all cases.
```

Comment:
```suggestion
    // The TTL for each key in seconds.
    //
    // In some TiKV of old versions, only one TTL can be provided and the TTL will be applied to all keys in 
    // this request. For compatibility reasons, if the length of `ttls` is exactly one, the TTL will be applied
    // to all keys. Otherwise, the length mismatch between `ttls` and `pairs` will return an error.
```